✨ Feat: Reservation Event 처리 채팅 연동 기능 추가

megaseunghan opened the pull request:
## 연관 이슈
- closes #122 

## 해당 작업을 추가/변경한 이유는 무엇인가요?
- WebSocket인 STOMP 프로토콜을 사용하여 예약 변경에 대한 내용을 전송하기 위함입니다.
- 이에 따라 메세지 전송을 위한 Manager가 따로 존재하는게 좋겠다고 생각하여 ChatMessageSendManager를 만들었습니다.

## 어떤 부분에 리뷰어가 집중하면 좋을까요?
- ADMIN 멤버에 대한 ID를 그대로 노출시키는게 맞나 싶습니다. 

## 리뷰어에게 추가/변경 내용을 상세히 설명해주세요!
- [x] MessageStatus 상태값 추가(NOTICE)
- [x] SchedulerService 채팅 전송 기능 추가
- [x] SimpMessageSendingOperations 통합(ChatMessageSendManager.kt)

megaseunghan committed:
✨ Feat: Reservation Event 처리 채팅 연동 기능 추가

- MessageStatus 상태값 추가(NOTICE)
- SchedulerService 채팅 전송 기능 추가
- SimpMessageSendingOperations 통합(ChatMessageSendManager.kt)

lovelyunsh commented on src/main/kotlin/com/challengeteamkotlin/campdaddy/application/scheduler/SchedulerService.kt:
```diff
@@ -9,20 +13,23 @@ import org.springframework.stereotype.Service
-            println(reservationEvent)
-
+                .run {
+                    val chatRoom = chatRoomRepository.getChatRoomByBuyerIdAndProductId(this.buyerId, this.productId)
+                        ?: throw EntityNotFoundException(ChatErrorCode.CHAT_NOT_FOUND)
+                    chatMessageSendManager.sendToChatRoom(chatRoom.id!!, it)
```
예약 생성 시에는 채팅방이 아직 존재 하지 않을 수 있을 것 같습니다.
예약 생성 할 때는 채팅을 안보내도록 정책을 바꾸거나,
채팅방이 없으면 채팅방을 생성하게 하거나 둘 중 하나로 변경 해야 할 것 같습니다.

megaseunghan replied:
좋네요! 한번 수정해보겠습니다

megaseunghan replied:
일단 후자로 수정커밋 올려볼게요

megaseunghan committed:
✨ Feat: SchedulerService 예약 최초 생성시 로직 수정

- 예약 최초 생성시, 채팅방이 없으면 채팅방을 생성하게 하는 로직 추가
- 관련하여 Product, Member Repository의 의존성 주입이 추가됨

megaseunghan committed:
✨ Feat: SchedulerService 예약 최초 생성시 로직 수정

- 예약 최초 생성시, 채팅방이 없으면 채팅방을 생성하게 하는 로직 재수정
- Repository의 주입보다, Service를 단방향으로 의존하여 처리하는 것이 코드가 깔끔해짐

megaseunghan committed:
✨ Feat: ChatMessageSendManager 오타 수정
